### ISSUE TITLE
Planning Questionnaire UI -> Main UI variable workflow(s)

### ISSUE THREAD

**Issue by eduardo-vie** [xlai/dosefinderhub#10]
- Depending on Questionnaire questions and their level of specificity, detail at least 2 potential variable workflows:
1. Questionnaire ultimately provides ranking of recommended trial designs + non-design-specific trial set-up information (e.g., maximum sample, number of doses, start dose, TTL) to main UI, which then collects design-specific info for simulations
2. Questionnaire ultimately **only** provides ranking to main UI, which collects both non-design-specific and design-specific trial set-up info for simulations

- This should include making ~3 variable lists (For Ranking, Non-Design-Specific, Design-Specific)

- Illustrate/summarize workflows in flow charts?

**Comment by eduardo-vie** [comment]
What did you use to make this? The Github comment box itself?

I'm using PPT and it's quite slow/annoying. :D

**Comment by eduardo-vie** [comment]
As per our catch-up on 17/8 afternoon, here is an updated high-level description of the dashboard workflow and display:
(I believe this is all we discussed. Just wanted to have it down somewhere (mostly for my own reference!)

**Questionnaire UI:**
- In first page, user will be asked whether they want to complete the Questionnaire or not. If YES, they will be directed to subsequent questionnaire questions. If NO, they we be sent straight to the Main UI.
- Subsequent questionnaire questions: first batch will be related to the User Profile (to be used to compute trial design recommendation ranking). Second batch will be non-design-specific parameters needed for the simulations (max sample size, no doses, start dose etc.).
- Based on that computed User Profile, Questionnaire UI will display a page with "flashcards" (little text boxes), each representing a design, ordered according to level of recommendation and featuring a short explanation. Users will be prompted to use a button in each box to specify design-specific parameters needed for each design's simulations. (Something I don't believe we directly discussed: regardless of the ranking's top 3, which 3 designs are to be simulated should be up to the user?)
- Parameters will flow to backend modelling. User will be directed to main UI.

**Main UI:**
- User will be directed to a Configurations tab. There, they will see 3 drop down columns, one for each simulated design. Once they click to drop down a column, if will be sectioned based on info type (clinical parameters, simulation parameters etc.). These sections will include the same trial-specific questions as in the latter portion of the Questionnaire. If the user has completed the Questionnaire, the previous answers will be fed into and displayed in this tab. If they have not, they will use this tab to provide this information.
- The user will then be able to flick through 3 other tabs, one for each simulated design, including main simulation and comparison outputs (DTPs, average no participants per dose, probabilities of dose selection etc.). It should also include some sort of summarized cross-design comparison output.
- The 5th tab will be a comparison tab, with more detailed cross-design comparison outputs.

**Comment by xlai** [comment]
thanks @eduardo-vie! I will update the mock-up shortly. Watch this space.

**Comment by eduardo-vie** [comment]
Currently working on a functional mock Conduct tab.
Some questions arose:
- Like the Design Tabs, should Conduct Tab(s) be triggered based on the ranking vector? In other words, should there always and only be Conduct Tabs for recommended designs?
- Should the Conduct Tabs include the method-specific inputs which have already been shown at the end of the Questionnaire and on the Configurations Tab, or should it just as for Conduct info (doses of previous cohorts, DLT rates of previous cohorts)?
- Differently from what I was coding, Jake suggested CRM Conduct info be collected in a reactive table. Each row would be a past (complete) cohort, and the columns would be dose level, cohort size and number of DLTs. I quite like this idea, but is there perhaps a better way?
- Tagging @xlai and @jakeholton for input :)

**Comment by xlai** [comment]
> Like the Design Tabs, should Conduct Tab(s) be triggered based on the ranking vector? In other words, should there always and only be Conduct Tabs for recommended designs?

In regards to the conduct tab, we can have only one tab but keep a left/right panel layout. Then implement a dropdown menu allowing user to select our top three methods as appropriate. [^1]

> Should the Conduct Tabs include the method-specific inputs which have already been shown at the end of the Questionnaire and on the Configurations Tab, or should it just ask for Conduct info (doses of previous cohorts, DLT rates of previous cohorts)?

Related to my first point, if the user were only allowed to choose between the top three methods, then you don't have to bring in everything. If user wants to change config, they should go to the config tab.

> Differently from what I was coding, Jake suggested CRM Conduct info be collected in a reactive table. Each row would be a past (complete) cohort, and the columns would be dose level, cohort size and number of DLTs. I quite like this idea, but is there perhaps a better way?

Is this only a specific CRM conduct issue or a general suggestion for all method? 

I would start with an empty table implement an add cohort button such that with each press:

- It will add number of rows that corresponds to the cohort size
- allow manual entering of dose level and dlt status (yes/no) separately for each patient in the cohort.
- add new column called 'evaluability' (yes/no) 
- 
Patient entry are sometimes staggered and not all the same time. We need to allow certain degrees of flexibility.

@jakeholton  when you read the dataframe for escalation, only read in rows that are complete and with evaluability marked 'yes'.
 

[^1]: the caveat being that the user won't be able to conduct the trial using any other method that weren't in the top three (for example)

**Comment by jakeholton** [comment]
I agree with the need for flexibility, which is why I suggested the reactive table rather than carrying over cohort_size.

A quick point on this table @xlai : would this be one single table? If so, might there be a need to identify which cohort a patient is in for some trial designs? I'd maybe suggest there be a single table that automatically adds 1 to a cohort number column on the far left. I'd also suggest it would need a level of editability with removing rows etc.

**Comment by christie94h** [comment]
Closing this issue as the app has a new structure (and therefore a new workflow). The new workflow of the app is discussed in issue #74.
